src/core/Akka/Util/ByteString.cs

```diff
@@ -191,6 +192,12 @@ public static ByteString FromString(string str, Encoding encoding)
         private readonly int _count;
         private readonly ByteBuffer[] _buffers;
 
+        public ByteString(IList<ByteString> buffers)
+        {
+            _count = buffers.Sum(s => s.Count);
+            _buffers = buffers.SelectMany(r=>r._buffers).ToArray();
+        }
+
```

Comment:
So, this is useful for some cases, but looking at it I can't help but wish we had a nice way to pack this and fetch lazily (so we don't have to do this SelectMany+ToArray